app/src/main/java/com/github/se/signify/ui/screens/profile/SettingsScreen.kt

```diff
@@ -88,135 +84,108 @@ fun SettingsScreen(
-                        placeholder = {
-                          Text(userName.value, color = MaterialTheme.colorScheme.onBackground)
-                        },
-                        modifier =
-                            Modifier.widthIn(max = 200.dp)
-                                .padding(vertical = 8.dp)
-                                .background(MaterialTheme.colorScheme.background),
-                        colors =
-                            TextFieldDefaults.colors(
-                                focusedContainerColor = MaterialTheme.colorScheme.background,
-                                unfocusedContainerColor = MaterialTheme.colorScheme.background,
-                                focusedTextColor = MaterialTheme.colorScheme.onBackground,
-                                cursorColor = MaterialTheme.colorScheme.onBackground))
-                  }
-            }
-
-        // Other Settings Section
-        Box(
-            modifier =
-                Modifier.fillMaxWidth()
-                    .clip(RoundedCornerShape(12.dp))
-                    .background(MaterialTheme.colorScheme.surface)
-                    .padding(16.dp),
-            contentAlignment = Alignment.Center) {
-              Text(
-                  text = "Other settings:\nLanguage,\n theme, ...",
-                  fontSize = 16.sp,
-                  color = MaterialTheme.colorScheme.onSurface,
-                  fontWeight = FontWeight.Normal)
-            }
-
-        // Cancel and Save Buttons
-        Row(horizontalArrangement = Arrangement.spacedBy(16.dp)) {
-          ActionButtons(
-              {
-                newName = ""
-                selectedImageUrl = profilePictureUrl.value
-              },
-              MaterialTheme.colorScheme.error,
-              "Cancel",
-              Modifier.weight(1f))
-
-          ActionButtons(
-              {
-                if (newName.isNotBlank()) {
-                  userViewModel.updateUserName(currentUserId, newName)
-                }
-                userViewModel.updateProfilePictureUrl(currentUserId, selectedImageUrl)
-              },
-              MaterialTheme.colorScheme.primary,
-              "Save",
-              Modifier.weight(1f))
+  AnnexScreenScaffold(navigationActions = navigationActions, testTagColumn = "SettingsScreen") {
+    // Editable Profile Picture
+    Column(modifier = Modifier.fillMaxWidth(), horizontalAlignment = Alignment.CenterHorizontally) {
+      Row(
+          modifier =
+              Modifier.border(
+                      2.dp, MaterialTheme.colorScheme.outlineVariant, RoundedCornerShape(8.dp))
+                  .clip(RoundedCornerShape(8.dp))
+                  .background(MaterialTheme.colorScheme.background)
+                  .padding(horizontal = 24.dp, vertical = 16.dp),
+          horizontalArrangement = Arrangement.Center,
+          verticalAlignment = Alignment.CenterVertically,
+      ) {
+        Column {
```

Comment:
As refactoring is the main purpose of the PR, you can add a new IconUtilis that does the same for the following icons

Reply:
You are right, that could be a future improvement with more util functions, but I can't do it now. So I suggest that it becomes a task for a next sprint.